R/generate_map_and_indicator.R

```diff
@@ -36,22 +35,62 @@
 #' @export
 
 generate_map_and_indicator <- function(
-    PD_cube,
+    pd_cube,
     grid,
     taxon = NULL,
     bbox_custom = NULL,
     cutoff = NULL) {
 
+  # Check that mcube is a dataframe or tibble
+  if (!is.data.frame(pd_cube)) {
+    stop("Error: 'pd_cube' must be a dataframe or tibble.")
+  }
+
+  if (!"pd" %in% colnames(pd_cube)) {
+    stop("`pd_cube` must contain a column named 'pd' representing Phylogenetic
+         Diversity.")
+  }
+
+  if (missing(grid) || !inherits(grid, "sf")) {
+    stop("`grid` must be provided and should be an `sf` object.")
+  }
+
+  if (!all(sf::st_is_valid(grid))) {
+    stop("Some geometries in `grid` are invalid. Please fix them before
+         proceeding.")
+  }
+
+  if (!"CELLCODE" %in% colnames(grid)) {
```

Comment:
stopifnot (and assertthat):

stopifnot("`grid` must contain a column named 'CELLCODE' for grid cell codes." = assertthat::has_name(grid, "CELLCODE"))